Amazon.QLDB.Driver/transaction/Transaction.cs

```diff
@@ -64,12 +64,15 @@ internal virtual void Abort()
         /// </exception>
         internal void Commit()
         {
-            byte[] hashBytes = this.qldbHash.Hash;
-            MemoryStream commitDigest = this.session.CommitTransaction(this.txnId, new MemoryStream(hashBytes))
-                .CommitDigest;
-            if (!hashBytes.SequenceEqual(commitDigest.ToArray()))
+            lock (this)
```

Comment:
Same thing, don't lock on something that can be access by the library consumer.

Reply:
Transaction.cs is not accessible by the library consumer.

Reply:
I think the point is that we should lock on an internal, see https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/lock-statement, specifically:

When you synchronize thread access to a shared resource, lock on a dedicated object instance (for example, `private readonly object balanceLock = new object();`) or another instance that is unlikely to be used as a lock object by unrelated parts of the code. Avoid using the same lock object instance for different shared resources, as it might result in deadlock or lock contention. In particular, avoid using the following as lock objects:

- `this`, as it might be used by the callers as a lock.
- Type instances, as those might be obtained by the `typeof` operator or reflection.
- `string` instances, including string literals, as those might be interned.

Hold a lock for as short time as possible to reduce lock contention.

Perhaps lock on this.qldbHash?